config/onprem-iso-fcc.yaml

```diff
@@ -129,3 +142,16 @@ storage:
           mode: 0644
           contents:
             inline: replace-with-your-urlencoded-pull-secret
+        - path: /etc/assisted-service/startup_script.sh
+          mode: 0644
+          contents:
+            inline: |
+              #!/bin/bash
+              ips=$(hostname -I)
+              read -r -a ipArr <<< "$ips"
+              for ip in "${ipArr[@]}"
+                  do 
+                      printf "\n%s assisted-api.local.openshift.io\n" "$ip" >> /etc/hosts
+                  done
+              ips_cs=`echo $ips | xargs | sed -e 's/ /,/g'`
+              printf "\nSERVICE_IPS=%s" "$ips_cs" >> /etc/assisted-service/environment
```

Comment:
I wish there was a way to test this :worried: 

Also, if I'm reading this correctly, we are going to set the same hostname to every available ip on the host...
That seems strange, is that really what we want to do?

Reply:
Yeah, it's a bit of a hack :exploding_head: We needed a way to get the IP of host running the service that is routable from the agent nodes, without making it yet another thing user has to provide. This will configure the discovery iso ignition with all the ips of the service host and the agent will try all the ips until it finds one that is routable form node where the agent iso is being booted.

Reply:
> the agent will try all the ips until it finds one that is routable form node where the agent iso is being booted

Where is this logic?

Reply:
I'm mostly concerned that this is being added to `/etc/hosts` and as far as I can tell we will only ever resolve the hostname to the first matching entry in the file. If we need this special logic I don't think we can use `/etc/hosts`.

Reply:
Nice, thanks @mhrivnak TIL.

Still kinda messy but makes sense now. :+1: